src/main/java/at/hannibal2/skyhanni/features/mining/DarkMonolithFeatures.kt

```diff
@@ -0,0 +1,174 @@
+    }
+
+    // Todo: need chat pattern for rock the fish drop
+    /**
+     * REGEX-TEST: §5§lMONOLITH! §r§aYou found a mysterious §r§5Dark Monolith §r§aand were rewarded §r§650,000 Coins§r§a!
+     * REGEX-TEST: §5§lMONOLITH! §r§aYou found a mysterious §r§5Dark Monolith §r§aand were rewarded §r§62,500 Coins §r§aand §r§21,000 ᠅ Mithril Powder§r§a!
+     * REGEX-TEST: §5§lMONOLITH! §r§aYou found a mysterious §r§5Dark Monolith §r§aand were rewarded §r§2100 ᠅ Mithril Powder§r§a!
+     * REGEX-TEST: §5§lMONOLITH! §r§aYou found a mysterious §r§5Dark Monolith §r§aand were rewarded §r§23,000 ᠅ Mithril Powder§r§a!
+     */
+    @Suppress("MaxLineLength")
+    private val dropPattern by patternGroup.pattern(
+        "drop",
+        "§5§lMONOLITH! §r§aYou.*§r§aand were rewarded ?(?:(?:§.)+(?<coins>[\\d,]+) Coins ?(?:§.)+)?(?:!|and )?(?:(?:§.)+(?<powder>[\\d,]+) ᠅ Mithril Powder§r§a!)?",
+    )
+
+    private var knownEggs: Set<LorenzVec> = setOf()
+    private var foundEggVec: LorenzVec? = null
+    private var lastFoundEggVec: LorenzVec? = null
+    private var renderBox: AxisAlignedBB? = null
+
+    private fun reset() {
+        knownEggs = setOf()
+        foundEggVec = null
+        lastFoundEggVec = null
+        renderBox = null
+    }
+
+    init {
+        RenderDisplayHelper(
+            outsideInventory = true,
+            inOwnInventory = true,
+            condition = { config.tracker && IslandType.DWARVEN_MINES.isCurrent() },
+            onRender = {
+                tracker.renderDisplay(config.trackerPosition)
+            },
+        )
+    }
+
+    @HandleEvent(onlyOnIsland = IslandType.DWARVEN_MINES)
+    fun onChat(event: SkyHanniChatEvent) {
+        dropPattern.matchMatcher(event.message) {
+            DarkMonolithFeatures.reset()
+            if (!config.tracker) return
+            groupOrNull("coins")?.let {
+                tracker.addCoins(it.formatInt(), false)
+            }
+            groupOrNull("powder")?.let {
+                tracker.addItem(mithrilPowderItem, it.formatInt(), false)
+            }
+            groupOrNull("fish")?.let {
+                tracker.addItem(rockTheFishItem, 1, false)
+            }
+            tracker.modify {
+                it.monolithsLooted++
+            }
+        }
+    }
+
+    @HandleEvent
+    fun onWorldChange(event: IslandChangeEvent) {
+        reset()
+        if (event.newIsland == IslandType.DWARVEN_MINES) {
+            tracker.firstUpdate()
+        }
+    }
+
+    @HandleEvent(onlyOnIsland = IslandType.DWARVEN_MINES)
+    fun onSecondPassed() {
+        if (!isEnabled()) return
+        knownEggs = BlockUtils.nearbyBlocks(
+            LocationUtils.playerLocation(),
+            distance = 40,
+            filter = Blocks.dragon_egg,
+        ).keys
+    }
+
+    @HandleEvent(onlyOnIsland = IslandType.DWARVEN_MINES)
+    fun onTick() {
+        if (!isEnabled()) return
+        foundEggVec = knownEggs.firstOrNull { it.canBeSeen() }
+        checkTitle()
```

Comment:
the check title call here should get moved above the is enabled check in the on tick function. we do not need to check if config.highlight is enabled for the titel at all